andela/jubilee-backend
# #167733645 Build out the User Login Endpoint

#### What does this PR do?
Have the user login endpoint working

#### Description of Task to be completed?
Have the following endpoint working 
POST `api/auth/login`

#### How should this be manually tested?
- Clone repo and change into the working directory `cd jubilee-backend`
- Switch to the **ft-user-login-167733645** branch and run `npm install`
- Create a `.env` file and specify the variables described in the `.env.sample` in your `.env`.
- Run `npm run migrate` to create tables in your database.
- Run `npm run start:dev` to start the development server.
- Launch Postman and make a POST request to the `signup route` http://localhost:3000/api/auth/signup using the fields described in the swagger document of the project. http://localhost:3000/api/docs or just simply run `npm run seed`
- Still on Postman, goto the `login route` http://localhost:3000/api/auth/login and make a POST request with the same email and password used in Signing up above
```javascript
{
   "email": "signupemail@jubille.com",
   "password": "signupPassword"
}

If you seeded the database with `npm run seed`, use this login details
{
   "email": "ade.steve@gmail.com",
   "password": "testing"
}
```
- A  response body with a generated token and user details should be returned as response

#### Any background context you want to provide?
Nil

#### What are the relevant pivotal tracker stories?
[#167733645](https://www.pivotaltracker.com/story/show/167733645)

#### Screenshots (if appropriate)
![login mockup](https://user-images.githubusercontent.com/36106823/63744943-c78cfa00-c898-11e9-8f8b-23ecf0e401c5.png)
